experimental/dds/tree2/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -940,15 +632,94 @@ function getIntention(
 	return rev === undefined ? undefined : revisionMetadata.getInfo(rev).rollbackOf ?? rev;
 }
 
-function areInverseRevisions(
-	rev1: RevisionTag | undefined,
-	rev2: RevisionTag | undefined,
-	revisionMetadata: RevisionMetadataSource,
-): boolean {
-	if (rev1 === undefined || rev2 === undefined) {
-		return false;
+// TODO: Try to share more logic with the version in rebase.ts.
+/**
+ * Returns a number N which encodes how the cells of the two marks are aligned.
+ * - If N is zero, then the first cell of `baseMark` is the same as the first cell of `newMark`.
+ * - If N is positive, then the first N cells of `newMark` (or all its cells if N is greater than its length)
+ * are before the first cell of `baseMark`.
+ * - If N is negative, then the first N cells of `baseMark` (or all its cells if N is greater than its length)
+ * are before the first cell of `newMark`.
+ */
+function compareCellPositions(
+	baseCellId: DetachEvent,
+	baseMark: ExistingCellMark<unknown>,
+	newMark: EmptyInputCellMark<unknown>,
+	newRevision: RevisionTag | undefined,
+	cancelledInserts: Set<RevisionTag>,
+	gapTracker: GapTracker,
+): number {
+	const newCellId = getCellId(newMark, newRevision);
+	if (baseCellId.revision === newCellId?.revision) {
+		if (isNewAttach(newMark)) {
+			// There is some change foo that is being cancelled out as part of a rebase sandwich.
+			// The marks that make up this change (and its inverse) may be broken up differently between the base
+			// changeset and the new changeset because either changeset may have been composed with other changes
+			// whose marks may now be interleaved with the marks that represent foo/its inverse.
+			// This means that the base and new marks may not be of the same length.
+			// We do however know that the all of the marks for foo will appear in the base changeset and all of the
+			// marks for the inverse of foo will appear in the new changeset, so we can be confident that whenever
+			// we encounter such pairs of marks, they do line up such that they describe changes to the same first
+			// cell. This means we can safely treat them as inverses of one another.
+			return 0;
+		}
+		return baseCellId.index - newCellId.index;
+	}
+
+	if (newCellId !== undefined) {
+		const offsetInBase = getOffsetAtRevision(baseMark.lineage, newCellId.revision);
```

Comment:
I find the name `offsetInBase` to be confusing: it sounds like the offset of something (though it's not clear what) in the revision of the base mark.
How about `baseOffsetInJointRevision` or just `baseOffset`?